pynuodb/session.py

```diff
@@ -197,14 +198,20 @@ def _parse_addr(self, addr, options):
                     if options['ipVersion'] == 'v6':
                         ver = 6
 
-                # get ip addr from hostname
-                af = socket.AF_INET
-                if ver == 6:
-                    af = socket.AF_INET6
-                try:
-                    ip = socket.getaddrinfo(host, port, af)[0][4][0]
-                except:
-                    raise SessionException("Addr lookup failed %s %d %d" % (host,port,af))
+                # get ip addr from hostname unless not running on Linux
+                # or verifyHostname is true
+                if platform.system() != 'Linux' or \
+                   (options != None and 'verifyHostname' in options \
+                    and strToBool(options['verifyHostname']) == True):
+                    ip = host
+                else:
+                    af = socket.AF_INET
+                    if ver == 6:
+                        af = socket.AF_INET6
+                    try:
+                        ip = socket.getaddrinfo(host, port, af)[0][4][0]
+                    except:
+                        raise SessionException("Addr lookup failed %s %d %d" % (host,port,af))
```

Comment:
I didn't look at this too closely the first time it was reviewed. This is eagerly doing DNS resolution if we are supplied a hostname? Is that necessary to enable IPv6? That would mean that IPv6 and TLS with server hostname verification can't work together.

Reply:
I found I needed the name resolution when running some cases of docker test. But notice it is disabled if tls hostname verification is enabled. That is what this change is about. Prior to this change it was always doing the name resolution, and that did cause failure noted in jira.

Reply:
So the eager DNS resolution (as opposed to just connecting to the hostname and letting DNS resolution happen automatically) is to enable testing?

Also, the default value is true (see http://doc.nuodb.com/Latest/Default.htm#nuoadmin-connection-properties.htm?Highlight=verifyHostname). So you should be checking the verifyHostname=False is not explicitly specified.

Added @vegichan so that he can give his input.

Reply:
The change was not to 'enable' testing. A test was failing when using hostname and expecting v6 resolution. The basic problem is that if a host has both v4 and v6 addresses, how do we select which address we want to use? I added a parameter to the python driver 'ipVersion' to allow user to select. Then the driver uses the parameter to set the address family and do a lookup. I can revisit to see if there is another, better way to do this. At the time I added this I was unaware of the tls issue; it showed up in the last couple days. Also will verify default; the code as is does seem to run ok on the bamboo test.

Reply:
> Also will verify default; the code as is does seem to run ok on the bamboo test.

That must be due to a gap in our test coverage. If TLS is enabled, which is determined by the presence of either the "trustStore" or "trustedCertificates" connection property, "verifyHostname" is assumed to be true unless explicitly set to false. Your code should work for non-TLS (almost all of our testing) and for TLS with explicit "verifyHostname" setting (I suspect all of our TLS testing), but I don't think it works for TLS without "verifyHostname" setting.

Reply:
I have decided to remove the hostname to ip addr conversion in all cases. Partly in the interest of not breaking something and partly because I am now unable to reproduce the problem that caused me to add it in the 1st place. Somehow now it seems that the underlying Python socket C library code is managing to get the right ip addr such that my ipv6-docker tests are working without the conversion in the driver. Perhaps future Ipv6 testing will uncover a problem but until it does would rather not cause something else to fail.